components/insight/SRC/org/openmicroscopy/shoola/env/data/login/LoginServiceImpl.java

```diff
@@ -183,7 +186,12 @@ private int attempt()
                     if (cause.getCause() instanceof SecurityViolation)
                         failureIndex = ACTIVE_INDEX;
                 }
-            } else failureIndex = SYSTEM_FAILURE_INDEX;
+                else if (exception.getMessage().contains("is not compatible")) {
```

Comment:
Maybe a `null` check first?

Reply:
May need a note in `Gateway` to update this if the message wording is changed.

Reply:
That's true, have to check for `null` first. It's a bit dodgy anyway parsing the message string to find out the specifics of the Exception, would maybe need another flag/field in the `DSOutOfServiceException` itself to account for that (will add a note to the gateway card).

Reply:
Could also subclass it.